micro-rdk/src/common/ota.rs

```diff
@@ -347,11 +394,25 @@ impl OtaService {
 
         #[cfg(feature = "esp32")]
         {
+            log::info!("setting device to use new firmware on reboot");
             update_handle.complete().map_err(OtaError::EspError)?;
-            log::info!("resetting now to boot from new firmware");
+        }
+
+        log::info!("updating metadata in NVS");
+        self.storage
+            .store_ota_metadata(OtaMetadata {
+                version: self.pending_version.clone()
+            })
+            .map_err(|e| OtaError::Other(e.to_string()))?;
+
+        log::info!("firmware update complete");
```

Comment:
@acmorrow @gvaradarajan 

in this section, there's the possibility we succeed in setting the next boot partition (`update_handle.complete()`) but fail to update NVS (`store_ota_metadata()`).

In this scenario, the device will boot from the new firmware but, given the old nvs metadata is still intact/corrupted/non-existant, will think it is out of date and attempt to update again (download firmware, set boot partition, update nvs).

This is similar to the initial boot-up anyway since there is no OtaMetadata in the NVS on a clean build.

It seems preferable to the alternative if we swap the order of updates; the device will (falsely) believeit is up-to-date and will not perform the update.